src/modules/place/PlaceView.js

```diff
@@ -0,0 +1,266 @@
+import React, {PropTypes, Component} from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Image,
+  Platform,
+  ActivityIndicator,
+  Dimensions,
+  ScrollView,
+  Linking
+} from 'react-native';
+import * as theme from '../../utils/theme';
+import * as Utils from '../../utils/utils';
+import Button from '../../components/Button';
+
+const window = Dimensions.get('window');
+
+class PlaceView extends Component {
+  static displayName = 'PlaceView';
+
+  static navigationOptions = {
+    title: 'Place',
+    header: {
+      tintColor: theme.colors.text,
+      style: {
+        backgroundColor: theme.colors.navBar
+      }
+    }
+  }
+
+  static propTypes = {
+    office: PropTypes.object.isRequired,
+    loading: PropTypes.bool.isRequired,
+    place: PropTypes.object.isRequired,
+    back: PropTypes.func.isRequired,
+    officeStateActions: PropTypes.shape({
+      retryPlace: PropTypes.func.isRequired
+    }).isRequired
+  };
+
+  componentWillMount() {
+    //if this.prop.place doesn't contain the right data go back to home screen
+    if (!this.props.place.name) {
+      this.props.back();
+    }
+  }
+
+  onNextPress = () => {
+    this.props.officeStateActions.retryPlace(this.props.office, this.props.place);
+  }
+
+  buildPhotosURL = () => {
+    var size = this.props.place.photos.groups[0].items.length;
+    var photo = this.props.place.photos.groups[0].items[Utils.getRamdonNumberBetweenRange(size, 0)];
+    return (photo.prefix + '500x500' + photo.suffix);
```

Comment:
If this is meant to control the loading of the size of the photos, usually we do pixel calculations based on the screen dimensions of the phone instead of arbitrary numbers. i.e. if you want it to take up about half the screen you'd load ~`window/2`  rounded to some integer.